enhancements/sig-architecture/63-hosted-addon/README.md

```diff
@@ -0,0 +1,262 @@
+  <tr>
+    <th class="tg-0pky" rowspan="2"><br>Label Value</th>
+    <th class="tg-c3ow">Default Mode</th>
+    <th class="tg-c3ow" colspan="2">Hosted Mode</th>
+    <th class="tg-c3ow" rowspan="2"><br>Example</th>
+  </tr>
+  <tr>
+    <th class="tg-c3ow">Managed Cluster</th>
+    <th class="tg-c3ow">Managed Cluster</th>
+    <th class="tg-c3ow">Hosting Cluster</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td class="tg-0pky"><span style="font-weight:400;font-style:normal">nonexist | hosted-mode-managed</span></td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow"></td>
+    <td class="tg-c3ow">CRD owned by the addon</td>
+  </tr>
+  <tr>
+    <td class="tg-0pky"><span style="font-weight:400;font-style:normal">hosted-mode-hosting</span></td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow"></td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow">deployment</td>
+  </tr>
+  <tr>
+    <td class="tg-0pky"><span style="font-weight:400;font-style:normal">hosted-mode-none</span></td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow"></td>
+    <td class="tg-c3ow"></td>
+    <td class="tg-c3ow"></td>
+  </tr>
+</tbody>
+</table>
+
+### For addon deployers
+
+Once the managed cluster is imported to the hub in `Hosted` mode, we propose to add an annotation 
+`addon.open-cluster-management.io/hosting-cluster-name` for `ManagedClusterAddon` to 
+indicate the addon should be deployed in Hosted mode, and the value of the annotation is the hosting cluster which 
+should:
+- be a managed cluster of the hub as well
+- be the same cluster where the managed cluster klusterlet(registration-agent & work-agent) runs
+
+#### How does the addon agent connect to the managed cluster?
+
+Addon agents deployments will be deployed in the `{ManagedClusterAddon.Spec.InstallNamespace}` namespeace, and the 
+addon deployer is required to provide a secret, which name is `{addon-name}-managed-kubeconfig` and contains the key 
+`kubeconfig` to connect to the managed cluster, in the `{ManagedClusterAddon.Spec.InstallNamespace}` namespace on the 
+hosting cluster.
+
+Some potential ways to provide the secret:
+- The addon deployer asks the managed cluster administrator for a specific permission kubeconfig, then create it on the 
+hosting cluster. Because the kubeconfig is generated by the managed cluster administrator, the addon developer does not 
+need to deploy any service-account, role, rolebinding on the managed cluster, labels these resources 
+`addon.open-cluster-management.io/manifest-type: hosted-mode-none`
+- The addon developer deploys service-account, role, rolebinding on the managed cluster, labels these resources 
+`addon.open-cluster-management.io/manifest-type: hosted-mode-managed`(or does not add this label, keep this label 
+non-exist), then develops a component to get the service account on the remote managed cluster, and retrieve the token 
+back to the hosting cluster to generate a kubeconfig.
```

Comment:
It seems like it would be very useful if this was done by the addon-framework (or it had functions to help with the process at least)

Reply:
Currently, I don't think there is a suitable component for this. This requires the component can connect to the managed cluster naturally and has access to get the service account secret back to the hosting cluster. It looks like only registration-agent and work-agent can meet the condition, but the feature is clearly not what they were designed for, the registration-agent is for registry a cluster/an addon to the HUB, and the work-agent is for applying manifestwork to a managed cluster.